system_metrics_collector/src/system_metrics_collector/periodic_measurement_node.cpp

```diff
@@ -0,0 +1,75 @@
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+
+#include "periodic_measurement_node.hpp"
+
+#include <chrono>
+#include <string>
+
+#include "rclcpp/rclcpp.hpp"
+
+
+PeriodicMeasurementNode::PeriodicMeasurementNode(
+  const std::string & name,
+  const std::chrono::milliseconds & measurement_period,
+  const std::string & publishing_topic)
+: Node(name),
+  measurement_period_(measurement_period),
+  publishing_topic_(publishing_topic)
+{}
+
+bool PeriodicMeasurementNode::setupStart()
+{
+  std::unique_lock<std::recursive_mutex> ulock(mutex);
+
+  if (!measurement_timer_) {
+    RCLCPP_DEBUG(this->get_logger(), "creating timer");
+
+    measurement_timer_ = this->create_wall_timer(
+      measurement_period_,
+      std::bind(&PeriodicMeasurementNode::performPeriodicMeasurement, this));
+
+  } else {
+    RCLCPP_WARN(this->get_logger(), "setupStart: measurement_timer_ already exists!");
+  }
+  return true;
+}
+
+bool PeriodicMeasurementNode::setupStop()
+{
+  std::unique_lock<std::recursive_mutex> ulock(mutex);
+
+  if (measurement_timer_) {
+    measurement_timer_->cancel();
+    measurement_timer_ = nullptr;
+  }
+  return true;
+}
+
+void PeriodicMeasurementNode::performPeriodicMeasurement()
+{
+  std::unique_lock<std::recursive_mutex> ulock(mutex);
+  periodicMeasurement();
```

Comment:
```suggestion
  periodicMeasurementLocked();
```

Reply:
I don't think we need this lock anymore: especially since the collection addition (via the moving average container) is lock protected.